adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/providers/TheThingsStackProvider.java

```diff
@@ -0,0 +1,186 @@
+
+package org.eclipse.hono.adapter.lora.providers;
+
+import java.util.Base64;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.hono.adapter.lora.GatewayInfo;
+import org.eclipse.hono.adapter.lora.LoraMessageType;
+import org.eclipse.hono.adapter.lora.LoraMetaData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A LoRaWAN provider with API for The Thing Stack.
+ */
+@Component
+public class TheThingsStackProvider extends JsonBasedLoraProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TheThingsStackProvider.class);
+
+    private static final String FIELD_THE_THINGS_STACK_ALTITUDE = "altitude";
+    private static final String FIELD_THE_THINGS_STACK_BANDWIDTH = "bandwidth";
+    private static final String FIELD_THE_THINGS_STACK_CODING_RATE = "coding_rate";
+    private static final String FIELD_THE_THINGS_STACK_DATA_RATE = "data_rate";
+    private static final String FIELD_THE_THINGS_STACK_DEV_EUI = "dev_eui";
+    private static final String FIELD_THE_THINGS_STACK_DOWNLINKS = "downlinks";
+    private static final String FIELD_THE_THINGS_STACK_END_DEVICE_IDS = "end_device_ids";
+    private static final String FIELD_THE_THINGS_STACK_EUI = "eui";
+    private static final String FIELD_THE_THINGS_STACK_F_CNT = "f_cnt";
+    private static final String FIELD_THE_THINGS_STACK_F_PORT = "f_port";
+    private static final String FIELD_THE_THINGS_STACK_FREQUENCY = "frequency";
+    private static final String FIELD_THE_THINGS_STACK_FRM_PAYLOAD = "frm_payload";
+    private static final String FIELD_THE_THINGS_STACK_GATEWAY_IDS = "gateway_ids";
+    private static final String FIELD_THE_THINGS_STACK_JOIN_ACCEPT = "join_accept";
+    private static final String FIELD_THE_THINGS_STACK_LATITUDE = "latitude";
+    private static final String FIELD_THE_THINGS_STACK_LOCATIONS = "locations";
+    private static final String FIELD_THE_THINGS_STACK_LONGITUDE = "longitude";
+    private static final String FIELD_THE_THINGS_STACK_LORA = "lora";
+    private static final String FIELD_THE_THINGS_STACK_RSSI = "rssi";
+    private static final String FIELD_THE_THINGS_STACK_RX_METADATA = "rx_metadata";
+    private static final String FIELD_THE_THINGS_STACK_SETTINGS = "settings";
+    private static final String FIELD_THE_THINGS_STACK_SNR = "snr";
+    private static final String FIELD_THE_THINGS_STACK_SPREADING_FACTOR = "spreading_factor";
+    private static final String FIELD_THE_THINGS_STACK_UPLINK = "uplink_message";
+    private static final String FIELD_THE_THINGS_STACK_USER = "user";
+
+    private static final String OBJECT_GATEWAYS = "gateways";
+    private static final String OBJECT_META_DATA = "metadata";
+
+    @Override
+    public String getProviderName() {
+        return "theThingsStack";
```

Comment:
why not `The Things Stack`?

Reply:
I don't really have a preference but it's the similar to `actiltyWireless` and `actilityEnterprise`.

Reply:
Ok, then let's stick with it